File: app/Models/DriverRequest.php
<?php

// app/Models/DriverRequest.php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class DriverRequest extends Model
{
    use HasFactory;

   protected $fillable = [
    'user_id',
    'driver_id',
    'description',
    'pickup_location',
    'dropoff_location',
    'status',
    'price',
];

public function user()
{
    return $this->belongsTo(User::class, 'user_id');
}

public function driver()
{
    return $this->belongsTo(Driver::class, 'driver_id');
}

}

File: lib/data/models/driver_request.dart
class DriverRequest {
  final int id;
  final int userId;
  final int? driverId; // optional, if you want to link driver
  final String description;
  final String? pickupLocation;
  final String? dropoffLocation;
  final String status;
  final double? price;
  final DateTime createdAt;
  final DateTime updatedAt;

  DriverRequest({
    required this.id,
    required this.userId,
    this.driverId,
    required this.description,
    this.pickupLocation,
    this.dropoffLocation,
    required this.status,
    this.price,
    required this.createdAt,
    required this.updatedAt,
  });

  factory DriverRequest.fromJson(Map<String, dynamic> json) {
    return DriverRequest(
      id: json['id'],
      userId: json['user_id'],
      driverId: json['driver_id'],
      description: json['description'],
      pickupLocation: json['pickup_location'],
      dropoffLocation: json['dropoff_location'],
      status: json['status'],
      price: json['price'] != null ? double.tryParse(json['price'].toString()) : null,
      createdAt: DateTime.parse(json['created_at']),
      updatedAt: DateTime.parse(json['updated_at']),
    );
  }

  Map<String, dynamic> toJson() => {
    'id': id,
    'user_id': userId,
    'driver_id': driverId,
    'description': description,
    'pickup_location': pickupLocation,
    'dropoff_location': dropoffLocation,
    'status': status,
    'price': price,
    'created_at': createdAt.toIso8601String(),
    'updated_at': updatedAt.toIso8601String(),
  };
}
